### ISSUE TITLE
Installation Fails with "error: command 'gcc' failed"

### ISSUE THREAD

**Issue by Mawiguk0** [wolflu05/inventree-cups-plugin#10]
Hi there,

did not find a solution so far, maybe you can help identify:

Inventree Installation with docker compose.
When I try to install inventree-cups-plugin from webui i get following error in log (see below)

I fixed this by adding following packages to the Dockerfile, I dont know exactly if that was due to something i missed, building the inventree image with your instructions:

```diff
diff --git a/Dockerfile.orig b/Dockerfile
index 8bd252b..cc950d3 100644
--- a/Dockerfile.orig
+++ b/Dockerfile
@@ -9,5 +9,9 @@ RUN apk add --no-cache cups-dev gcc git musl-dev && \
 # production image - only install the cups shared library
 FROM inventree/inventree:${INVENTREE_TAG} as production

-RUN apk add --no-cache cups-libs
+RUN apk add --no-cache \
+       cups-libs \
+       cups-dev \
+       gcc \
+       musl-dev
 COPY --from=prebuild /root/.local /root/.local
```

so that the Dockerfile looks like this:
```Dockefile
ARG INVENTREE_TAG

# prebuild stage - needs a lot of build dependencies
FROM python:3.10-alpine3.18 as prebuild

RUN apk add --no-cache cups-dev gcc git musl-dev && \
    pip install --user --no-cache-dir git+https://github.com/wolflu05/inventree-cups-plugin

# production image - only install the cups shared library
FROM inventree/inventree:${INVENTREE_TAG} as production

RUN apk add --no-cache \
        cups-libs \
        cups-dev \
        gcc \
        musl-dev
COPY --from=prebuild /root/.local /root/.local
```

Error Log from docker logs

```bash
inventree-server  | Command '['/usr/local/bin/python', '-m', 'pip', 'install', '-U', 'inventree-cups-plugin']' returned non-zero exit status 1.
inventree-server  | /root/.local/lib/python3.11/site-packages/django/db/models/fields/__init__.py:1535: RuntimeWarning: DateTimeField NotificationEntry.updated received a naive datetime (2024-04-09 00:00:00) while time zone support is active.
inventree-server  |   warnings.warn(
inventree-server  | Pip command failed: Collecting inventree-cups-plugin
inventree-server  |   Downloading inventree_cups_plugin-0.2.0-py3-none-any.whl.metadata (4.7 kB)
inventree-server  | Collecting pycups (from inventree-cups-plugin)
inventree-server  |   Downloading pycups-2.0.1.tar.gz (62 kB)
inventree-server  |      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.0/63.0 kB 3.1 MB/s eta 0:00:00
inventree-server  |   Preparing metadata (setup.py): started
inventree-server  |   Preparing metadata (setup.py): finished with status 'done'
inventree-server  | Downloading inventree_cups_plugin-0.2.0-py3-none-any.whl (6.0 kB)
inventree-server  | Building wheels for collected packages: pycups
inventree-server  |   Building wheel for pycups (setup.py): started
inventree-server  |   Building wheel for pycups (setup.py): finished with status 'error'
inventree-server  |   error: subprocess-exited-with-error
inventree-server  |
inventree-server  |   × python setup.py bdist_wheel did not run successfully.
inventree-server  |   │ exit code: 1
inventree-server  |   ╰─> [1 lines of output]
inventree-server  |       error: command 'gcc' failed: No such file or directory
inventree-server  |       [end of output]
inventree-server  |
inventree-server  |   note: This error originates from a subprocess, and is likely not a problem with pip.
inventree-server  |   ERROR: Failed building wheel for pycups
inventree-server  |   Running setup.py clean for pycups
inventree-server  | Failed to build pycups
inventree-server  | ERROR: Could not build wheels for pycups, which is required to install pyproject.toml-based projects
inventree-server  |
inventree-server  | Bad Request: /api/plugins/install/
```

**Comment by wolflu05** [comment]
Hi, if you followed the docker instructions, you normally don't need to install it manually afterwards through the webui. The plugin should already be installed on your InvenTree instance as part of the newly build docker image. You just need to enable it after the first start in the plugin settings.

Please let me know if that works.

**Comment by Mawiguk0** [comment]
Currently trying to replicate the behavior (10min)

**Comment by Mawiguk0** [comment]
Btw. because im currently struggling to get the dymo LW 550 to work with inventree.

The hassle with getting cups to recognize dymo is already done. I can print the Testpages and with windows over network.
(Pi running cups, Dymo attached with usb to the pi)
The Installation of your cups plugin is running as well.

But if i now want to create a machine in the /platform UI it doesnt seem right.

Can you give me another hint/ressource how to print the labels with cups/dymo/inventree?

**Comment by wolflu05** [comment]
> Btw. because im currently struggling to get the dymo LW 550 to work with inventree.

I personally started to find a solution to print labels with my LW450 duo from inventree. For the first year cups was the only thing I could find to print with that printer. But using that approach has a few downsides for me. 
- I'm using the LW450 printer attached to a raspberry pi zero w running cups on nixos, which is really slow, because the rpi0 has to the work of converting the pdf to a data stream which can be sent to the printer (which takes around 3s).
- Creating an own image just to install this native cups dependency bothered me since a while.

So I started to investigate a few months ago if there is any other solution, and came around a technical documentation for the **LW4XX** series, so I wrote this https://github.com/wolflu05/inventree-dymo-plugin plugin which directly transforms the pdf to a data stream which can be interpreted by dymo (I now use cups just as a network printer queue server, you can do that by configuring the raw device like shown it the blogpost that I linked on the readme on how to setup cups).  But unfortunately this is not compatible with the LW5XX series because as far as I know they use a different protocol.

What I read about the LW5XX series was that it has a big limitation of label media you can use, because it actually checks if its original dymo media or not, and only can print with certified labels. 

> But if i now want to create a machine in the /platform UI it doesnt seem right.

What exactly do you mean here? Please provide more details.
(Do you specifically mean this error? https://github.com/wolflu05/inventree-dymo-plugin/issues/2#issuecomment-2010271005 )

**Comment by Mawiguk0** [comment]
Thank you for your help so far.
The LW5xx drivers from dymo are a bit different, but i managed to compile them with their, somewhat odd repository:
https://github.com/dymosoftware/Drivers/issues/3#issuecomment-2047071310

>What exactly do you mean here? Please provide more details.
That is actually already kind of solved, because it is running so far. I would like to investigate the behavior further once the printer is running.

Situation now is:

I thought, that if cups is now running with the LW550 and cups beeing connected to InvenTree, that there is little work left. Here are some screenshots.

![image](https://github.com/wolflu05/inventree-cups-plugin/assets/17619897/35a53aec-9dc0-4c58-b99e-6be76c308c9b)
![image](https://github.com/wolflu05/inventree-cups-plugin/assets/17619897/92f08d22-d4c7-4122-8601-b0911be93f89)
But the problem is, that I cannot select the printer in the print label modal/form:
![image](https://github.com/wolflu05/inventree-cups-plugin/assets/17619897/52bf2d37-77b5-4531-92fd-6d9399c8f62d)

**Comment by Mawiguk0** [comment]
And another question regarding the Plugin Configuration? 
Is it intended, that I have to put the key-values into the databse with psql or is there a menu that I so far have no discovered yet?

**Comment by Mawiguk0** [comment]
![image](https://github.com/wolflu05/inventree-cups-plugin/assets/17619897/911ffcd5-8348-44b5-863f-0c7680c282ed)
Now it is showing something i think might be correct. but still no labels beeing printed

**Comment by wolflu05** [comment]
Yes, machines should be configured through the dialog from your last screenshot.

All looks fine now. But regarding printing. Do I understand it right, that the machine showed up in the dialog, but when hitting print nothing happens? Could you take a look at the logs? Do the show something similar to https://github.com/wolflu05/inventree-dymo-plugin/issues/2#issuecomment-2010271005 ? Do you run a InvenTree worker?

**Comment by Mawiguk0** [comment]
Yes this is the problem, you tailored with the PR. Can i easilly disable workers?

> Do the show something similar to https://github.com/wolflu05/inventree-dymo-plugin/issues/2#issuecomment-2010271005 ?


it shows exactly the same...

**Comment by Mawiguk0** [comment]
Okey so I now follow all regarding issues and your PR.
I´m looking forward to see progress. If i can do anything for your PR let me know.

Thank you for your support so far! ❤️

**Comment by wolflu05** [comment]
This should be fixed now with inventree>=0.16.0 (Is currently scheduled to be released in a month or so) and this plugin in version >=1.0.0. (Note that v1.0.0 of this plugin will only work with Inventree>=0.16.0).

And regarding your initial docker issue, I guess this should be resolved now if you change the following (also updated that in the readme):
```diff
-FROM python:3.10-alpine3.18 as prebuild
+FROM python:3.11-alpine3.18 as prebuild
```

**Comment by Mawiguk0** [comment]
Thank you so much for the effort!

I currently want to migrate to a new host and the installation of inventree gets stuck at the static folder during invoke update, but that is something for the inventree issues^^

**Comment by Mawiguk0** [comment]
would this work for testing with eventula tag latest?

**Comment by wolflu05** [comment]
Yes, the fix from InvenTree/inventree#6772 is already merged in the master branch, so inventree/inventree:latest (which contains version 0.16 dev) should work just fine. (You may need to again pull the image, as you may already have pulled an older version tagged with latest). And note that a global cache is required (you need to enable redis).
